lib/fileset/default.nix

```diff
@@ -276,6 +278,45 @@ If a directory does not recursively contain any file, it is omitted from the sto
         _unionMany
       ];
 
+  /*
+    The file set containing all files that are in both of two given file sets.
+    See also [Intersection (set theory)](https://en.wikipedia.org/wiki/Intersection_(set_theory)).
+
+    The given file sets are evaluated as lazily as possible,
+    with the first argument being evaluated first if needed.
+
+    Type:
+      intersection :: FileSet -> FileSet -> FileSet
+
+    Example:
+      # Limit the selected files to the ones in ./., so only ./src and ./Makefile
+      intersection ./. (unions [ ../LICENSE ./src ./Makefile ])
```

Comment:
For a more realistic example that readily applies in many projects:

```nix
let buildFiles = fileFilter (f: f.ext != "nix") ../.;
in intersection buildFiles (unions [ ../build-support ./src ./Makefile ])
```

Not sure if I got the filter right. Had to change `LICENSE` in order to realistically encounter `.nix` files in the union.

Reply:
Initially I wrote the `fileFilter` without a path (`../.`). This would have been feasible if in a design where we allow infinite, non-enumerable sets such as `intersections [ ]`.

Reply:
`buildFiles` is particularly useful if you have multiple derivations that each build some part of that set.